ansible/roles_ocp_workloads/ocp4_workload_pipelines/readme.adoc

```diff
@@ -0,0 +1,66 @@
+= ocp4_workload_pipelines - Deploy OpenShift Pipelines to an OpenShift Cluster
+
+== Role overview
+
+* This role installs OpenShift Pipelines (backed by Tekton) into an OpenShift Cluster. It consists of the following playbooks:
+** Playbook: link:./tasks/pre_workload.yml[pre_workload.yml] - Sets up an
+ environment for the workload deployment.
+*** Debug task will print out: `pre_workload Tasks completed successfully.`
+
+** Playbook: link:./tasks/workload.yml[workload.yml] - Used to deploy OpenShift Pipelines
+*** This role creates a namespace (project) and deploys the operator. It will then create the Pipelines Install object that deploys the controllers etc.
+*** Debug task will print out: `workload Tasks completed successfully.`
+
+** Playbook: link:./tasks/post_workload.yml[post_workload.yml] - Used to
+ configure the workload after deployment
+*** This role doesn't do anything here
+*** Debug task will print out: `post_workload Tasks completed successfully.`
+
+** Playbook: link:./tasks/remove_workload.yml[remove_workload.yml] - Used to
+ delete the workload
+*** This role removes OpenShift Pipelines
+*** Debug task will print out: `remove_workload Tasks completed successfully.`
+
+== Review the defaults variable file
+
+* This file link:./defaults/main.yml[./defaults/main.yml] contains all the variables you need to define to control the deployment of your workload.
+* The variable *ocp_username* is mandatory to assign the workload to the correct OpenShift user.
+* A variable *silent=True* can be passed to suppress debug messages.
+
+
+=== Deploy a Workload with the `ocp-workload` playbook [Mostly for testing]
+
+----
+TARGET_HOST="bastion.ocp44.openshift.opentlc.com"
+OCP_USERNAME="wkulhane-redhat.com"
+WORKLOAD="ocp4_workload_pipelines"
+GUID=1001
+
+# a TARGET_HOST is specified in the command line, without using an inventory file
+ansible-playbook -i ${TARGET_HOST}, ./configs/ocp-workloads/ocp-workload.yml \
+    -e"ansible_ssh_private_key_file=~/.ssh/keytoyourhost.pem" \
```

Comment:
That is a "legacy" way of loading the workload. It feels a bit hacky and not in the vein of how we generally use config in agnosticd.

We have a config, ocp-workloads that does that and you don't have to touch the inventory. So it goes like this:

```yaml
cloud_provider: none
env_type: ocp-workloads

ocp_workloads:
  - ocp4_workload_pipelines

target_host: bastion.dev.mycluster.mydomain.com

#become_override: false

# If the ocp-workload supports it, you should specify the OCP user:
# ocp_username: myuser

# Usually the ocp-workloads want a GUID also:
# guid: changeme
```

Or target host as a dictionary:

```yaml
----
target_host:
  ansible_host: bastion.dev.mycluster.mydomain.com
  ansible_port: 22
  ansible_user: ec2-user
  #ansible_ssh_private_key_content: "{{ ssh_private_key_content }}"
  ansible_ssh_private_key_file: ~/.ssh/admin_key.pem
  #ansible_ssh_extra_args:  ...
```

And then  `ansible-playbook main.yml -e @vars.yml`

Reply:
That's cute and all. But easier it is not...